FogCreek/Glitch-Community
# Superb guide: Allow experiments to test against user signed in and project status

## Links
https://superb-guide.glitch.me/secret
https://app.clubhouse.io/glitch/story/5135/optimizely-features-that-depend-on-the-user-being-signed-out-and-not-having-projects

## Changes:
- Added a new 'user has projects' cookie to track if the user has projects for server rendering
- Send the hasLogin and hasProjects to optimizely so experiments can be made conditional on them

## How To Test:
Reload the secret page and run `window.cookie='optimizely-id='` in an incognito window until the swap test is enabled, then copy your persistent token to glitch.com and create a project. When you refresh the secret page the experiment should not be active.

## Things left to do before deploying:
- Get the hasProjects cookie whitelisted in cloudfront (not needed until the homepage/create test)
